x/wasm/internal/types/types.go

```diff
@@ -53,20 +54,22 @@ func (c CodeInfo) ValidateBasic() error {
 }
 
 // NewCodeInfo fills a new Contract struct
-func NewCodeInfo(codeHash []byte, creator sdk.AccAddress, source string, builder string) CodeInfo {
+func NewCodeInfo(codeHash []byte, creator sdk.AccAddress, source string, builder string, accessType AccessType) CodeInfo {
 	return CodeInfo{
-		CodeHash: codeHash,
-		Creator:  creator,
-		Source:   source,
-		Builder:  builder,
+		CodeHash:          codeHash,
+		Creator:           creator,
+		Source:            source,
+		Builder:           builder,
+		InstantiateConfig: accessType.With(creator),
 	}
 }
 
 // ContractInfo stores a WASM contract instance
 type ContractInfo struct {
-	CodeID  uint64          `json:"code_id"`
-	Creator sdk.AccAddress  `json:"creator"`
-	Admin   sdk.AccAddress  `json:"admin,omitempty"`
+	CodeID       uint64         `json:"code_id"`
+	Creator      sdk.AccAddress `json:"creator"`
+	ModifyConfig AccessConfig   `json:"modify_config"`
```

Comment:
Okay, I don't see why we would ever want "Everybody" here, which why you broaden it. So this is not incorrect, but maybe a bit more verbose API than desired.

Reply:
I don't see a use case for "Everybody" here other than somebody intentionally wanting to loose control over a contract.

Reply:
If we ever want AccessConfig to contain a list of addresses or do other extensions in this field the re usage of AccessConfig makes sense. But I see your point

Reply:
>  If we ever want AccessConfig to contain a list of addresses or do other extensions in this field the re usage of AccessConfig makes sense. But I see your point

Okay. If the option was `Nobody | Somebody(Addr) | Somebodies(Addr[]) | Everybody` then I see your point that "all options - Everybody" != `Nobody | Somebody`.

Reply:
I would like to keep things simplest possible given the functionality. That means either:

* Making this one `Admin` field like before (which satisfies the current functionality) OR
* Extend the `AccessConfig` with another option now that make this required (please think a bit on what it should be)

Reply:
> I don't see a use case for "Everybody" here other than somebody intentionally wanting to loose control over a contract.

This is just chaos. And much more likely to be a typo/misconfiguration than an honest request

Reply:
Let's keep things simple with the admin field. The changes are minimal on the server side and whenever we want to support multiple addresses we can do it easily. 👌